src/System.Runtime.Environment/src/OSName.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSName : IEquatable<OSName>
+    {
+        private string _name;
+
+        private const string WindowsName = "windows";
```

Comment:
On that note, should be be overriding ToString? Otherwise ToString won't return anything useful ("System.Runtime.InteropServices.OSName" I think).